datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachService.java

```diff
@@ -106,19 +131,33 @@ public void reattachStack(SdxCluster cluster, String originalName) {
     /**
      * Throws a NotFoundException if the database is not found for the cluster CRN and environment CRN.
      */
-    private void reattachExternalDatabase(SdxCluster cluster, String detachedCrn, String detachedName) {
+    private void reattachExternalDatabase(SdxCluster cluster, String detachedCrn) {
         LOGGER.info("Started reattaching external database for SDX cluster with ID: {}", cluster.getId());
+        sdxAttachDetachUtils.updateExternalDatabase(cluster, detachedCrn);
+        LOGGER.info("Finished reattaching external database for SDX cluster with ID: {}. Now has crn: {}.",
+                cluster.getId(), cluster.getCrn());
+    }
 
-        try {
-            sdxAttachDetachUtils.updateExternalDatabase(cluster, detachedCrn);
-            LOGGER.info("Finished reattaching external database for SDX cluster with ID: {}. Now has crn: {}.",
-                    cluster.getId(), cluster.getCrn());
-        } catch (RuntimeException e) {
-            // Undo the stack update to put everything back in its original state.
-            sdxAttachDetachUtils.updateStack(cluster.getClusterName(), detachedName, detachedCrn, true);
-            jobService.schedule(cluster.getId(), SdxClusterJobAdapter.class);
-            throw e;
-        }
```

Comment:
Why are you changing this?

Reply:
Yes this is refactoring. I essentially created a new method which will revert the stack update if an exception occurs during reattach.
The reason I do this is that with the new cluster proxy step we need the same logic so I decided to just have it all centralized in one method.